deepspeed/runtime/pipe/engine.py

```diff
@@ -926,15 +934,17 @@ def _send_tensor_meta(self, buffer, recv_stage):
                 * ndims
                 * shape
         """
-        send_bytes = 0
+        meta_buffer = torch.empty(TENSOR_META_SIZE, dtype=torch.int32, device=self.device)
```

Comment:
How can we guard against overflow if entries into `meta_buffer` is more than `TENSOR_META_SIZE`

Reply:
I added assertions to check the buffer size